plugin/keyring_vault/vault_key.h

```diff
@@ -0,0 +1,38 @@
+#ifndef MYSQL_VAULT_KEY_H
+#define MYSQL_VAULT_KEY_H
+
+#include "keyring_key.h"
+#include "i_serialized_object.h"
+
+namespace keyring {
+
+struct Vault_key : public Key, public ISerialized_object
+{
+  Vault_key(const char *a_key_id, const char *a_key_type, const char *a_user_id,
+      const void *a_key, size_t a_key_len)
+    : Key(a_key_id, a_key_type, a_user_id, a_key, a_key_len)
+    , was_key_retrieved(FALSE)	
+  {}
+
+  Vault_key(const Vault_key &vault_key)
+    : Key(vault_key.key_id.c_str(), vault_key.key_type.c_str(),
+          vault_key.user_id.c_str(), vault_key.key.get(), vault_key.key_len)
+  {
+    this->key_operation = vault_key.key_operation;
+    this->was_key_retrieved = FALSE;
+  }
+  Vault_key()
+  {}
+
+  virtual my_bool get_next_key(IKey **key);
+  virtual my_bool has_next_key();
+  virtual void create_key_signature() const;
+  void xor_data();
+
+protected:
```

Comment:
I am having problem with private inheritance in keyring framework. The same argument could be used for keyring::Key - when I developed it - there were no class inheriting from it. However, if I used only private instead of protected it would be really hard for me to inherit from Key class in Vault_key class.